### ISSUE TITLE
setError() should have an option to use a theme property

### ISSUE THREAD

**Issue by oculos** [keycloak/keycloak#23552]
### Description

Hi,

Since messages in a theme, defined on the `theme.properties` and `message_locale`, can be customized, it would be nice if the `setError()` method of a form could be configured to display a theme message. That way, errors could be displayed in different languages.

### Motivation

This would make creating custom Authenticators and other SPI's easier, especially in instances that have a lot in many languages.

**Comment by jonkoops** [comment]
Agreed, this should be localizable. Do we currently have any such localization anywhere else that we can use as inspiration?

**Comment by oculos** [comment]
> Agreed, this should be localizable. Do we currently have any such localization anywhere else that we can use as inspiration?

I think that a good alternative would be something like creating a new method, such as `setErrorThemeProperty(String property)`, and create the corresponding tag on the ftl template that would read the property as an error.

**Comment by ssilvert** [comment]
Link to keycloak-dev discussion:
https://groups.google.com/g/keycloak-dev/c/_Bys5pn-7qY/m/tPhuIpw_BwAJ

**Comment by ssilvert** [comment]
@oculos I'm going to take this out of triage.  But if you really want to see it implemented any time soon, we will probably need a PR from you or someone else in the community.

**Comment by ssilvert** [comment]
@oculos For localization, you usually don't need to test a message to see if it is a key.  You just pass in what might be key.  If the key is found then you will get the localized value back.  If the key is not found that means it's probably just a literal, so it just passes back the key.  At least that's how these APIs are usually designed.

If you want to point me to the code you are talking about, I might be able to provide some hints.

**Comment by oculos** [comment]
> @oculos For localization, you usually don't need to test a message to see if it is a key. You just pass in what might be key. If the key is found then you will get the localized value back. 

Yes, I know that. But the thing is that the way `template.ftl` is today, it simply shows the `message.summary` string - it does not use the `message.summary` as a key (https://github.com/keycloak/keycloak/blob/4291f4b01bd568f98531dc5754fcc0195c56f222/themes/src/main/resources/theme/base/login/template.ftl#L133) 

My idea is to test if `message` has an attribute to so that under `template.ftl` we could choose to either interpret `message` as a String to be shown or as a key.

Is it a bit clearer now?

**Comment by oculos** [comment]
@ssilvert Did you have a chance to see my answer? I am willing to work on this but would love to be pointed to the right direction :)

**Comment by oculos** [comment]
> It should be fine running `${msg(message.summary)}`

What do you mean @jonkoops ? The idea here is to send a key to a message that's part of the theme, so that it can be localized. `message.summary` will print the literal text, no?

**Comment by jonkoops** [comment]
Not sure what you need here. But if you want to localize some text it should run through that function. Like @ssilvert mentioned, we don't care if that is a full message, or a key for a translation, as it will fall back to whatever text was provided to `msg` when a translation cannot be found.
